src/threadgroup.h

```diff
@@ -33,6 +33,17 @@ class thread_group
         }
         threads.clear();
     }
+
+    // In the destructor, detach any threads that are still running.
+    // We expect that you've cleaned them up if you want to, probably via join_all()
```

Comment:
I think you should warn here. You can be expected to explicitly detach if that is what you want.

Reply:
I think there's even an argument for DbgAssert(...) or perhaps the destructor should automatically join all threads.  I am going to merge this to move the release forward, but let's discuss on tuesday what we want the behavior to be -- perhaps do some research on what other thread group abstractions do.